### ISSUE TITLE
After switching to HTTPS, Apache statistics stopped working

### ISSUE THREAD

**Issue by borzaka** [mikaku/Monitorix#252]
Today I switched to HTTPS, and permanently redirected all traffic to the `https://` variant.

Later I noticed, that the Apache statistics stopped gathering data:
![image](https://user-images.githubusercontent.com/1230402/64899214-a4c95680-d68a-11e9-8680-b7073958c3c1.png)

### Basic info:
- Debian 10.1
- Monitorix v3.11.0 (updated from this repo)
- Apache 2.4.38
- I used this tutorial: [How To Create a Self-Signed SSL Certificate for Apache in Debian 10](https://www.digitalocean.com/community/tutorials/how-to-create-a-self-signed-ssl-certificate-for-apache-in-debian-10)
- I don't have a domain yet, just an IP address for my server.

My Apache Monitorix settings looks like this: probably a default one
```
# APACHE graph
# -----------------------------------------------------------------------------
<apache>
        list = http://localhost/server-status?auto
        <alerts>
        </alerts>
        rigid = 0, 0, 2, 0, 0, 0
        limit = 100, 100, 100, 100, 100, 100
</apache>
```

**Comment by mikaku** [comment]
Hi, and thanks for your detailed information.

In order to know if the problem is in Monitorix or in the Apache itself, please check your Apache with the following command:

```
curl http://localhost/server-status?auto
```

and let me know if your receive something like this:

```
Total Accesses: 95425
Total kBytes: 4827997
CPULoad: .0289257
Uptime: 455131
ReqPerSec: .209665
BytesPerSec: 10862.5
BytesPerReq: 51808.9
BusyWorkers: 1
IdleWorkers: 19
Scoreboard: __W__.________.....____.___.....................................................................................................................................................................................................................................
```

Thanks.

**Comment by borzaka** [comment]
I get this:
```
$ curl http://localhost/server-status?auto
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="https://x.x.x.x/server-status?auto">here</a>.</p>
<hr>
<address>Apache/2.4.38 (Debian) Server at localhost Port 80</address>
</body></html>
```

If I change to `https`:
```
$ curl https://localhost/server-status?auto
curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

For now, I have changed to `https` in the `monitorix.conf`, restarted Monitorix, but still no data. `accept_selfsigned_certs` is set to yes.

**Comment by mikaku** [comment]
I don't see a proper output of Apache stats even with `curl`. Please, make sure to fix this with `curl` and then we'll try to also fix it in Monitorix.

**Comment by IzzySoft** [comment]
@borzaka The issue is that there's no proper cert for localhost. You need to exclude localhost from being redirected.

@mikaku nothing you can fix in Monitorix here. That's plain Apache config stuff.

**Comment by borzaka** [comment]
Then what's the point of the `accept_selfsigned_certs` setting in Monitorix if not accepting self signed certs even for localhost?

**Comment by mikaku** [comment]
@borzaka, I sill don't see stats using the `curl` command. Please, demonstrate first that your Apache is well configured to show stats.

Also, check in the log file to see if there are some errors.

@IzzySoft, thanks for chiming in. Let's see if `curl` can really get stats from his Apache.

**Comment by IzzySoft** [comment]
Yes, let's see. It's just that `https:` doesn't make much sense for `localhost` – because, whom you're kidding there? Who sits beween `127.1` and `localhost` you need to encrypt and hide traffic from? Of course one *could* do that. I just don't see it makes much sense. And right: that can only be achieved via self-signing – as the cert would be valid for *any* host (which again "degrades" its usefulness).

**TL;DR:** Unless there's a reason I miss, it's not worth it putting much effort into.

**Comment by borzaka** [comment]
Sorry, I'am unable to switch off the https redirect on localhost. So it wont produce any stats with `curl`, because it wont accept self signed certificate. I'am stuck.

I have this in my `sites-available/000-default.conf`:
`Redirect permanent "/" "https://IP-OF-MY-SERVER/"`

**Comment by IzzySoft** [comment]
Try making a `<VirtualHost:80>` (without the redirect) for `ServerName: localhost`. Or make the redirect conditional, based on `ServerName` – but I currently lack the time to write that up (check `RedirectMatch` in the Apache docs).

**Comment by borzaka** [comment]
I was able to make it work booth ways:
- new `<VirtualHost *:80>` with `ServerName` `localhost` and no `Redirect`
- same `<VirtualHost *:80>` with `RedirectMatch`
`RedirectMatch permanent ^(/(?!server-status).*)$ "https://IP-OF-MY-SERVER$1"`

I wanted to make the `RedirectMatch` with except `localhost`, but I couldn't write a matching regexp, so I ended up searching a phrase in the url, which is looks like this:
http://localhost/server-status?auto

So, with one of these modifications no https redirects on localhost, so no certificate problem, curl working.

Still I don't get the `accept_selfsigned_certs` setting in Monitorix. What's the purpose if not accepting self signed certs as was in my case?

Maybe this worth a note in the docs to setup Apache2 correctly if you have redirects to https.

**Comment by mikaku** [comment]
> Still I don't get the `accept_selfsigned_certs` setting in Monitorix. What's the purpose if not accepting self signed certs as was in my case?

The option `accept_selfsigned_certs` affects the behavior of the attribute `ssl_opts` in the [LWP::UserAgent](https://metacpan.org/pod/LWP::UserAgent#ssl_opts) Perl module. It basically sets the option `verify_hostname` to zero as stated in [here](https://metacpan.org/pod/LWP::UserAgent#verify_hostname-=%3E-$bool).

Glad to know you managed to solve the configuration problems in your Apache.
